Source/Synthesis.Solr/ContentSearch/SynthesisSolrFieldNameTranslator.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Sitecore.ContentSearch;
+using Sitecore.ContentSearch.Linq.Common;
+using Sitecore.ContentSearch.SolrProvider;
+using Sitecore.Data;
+using Sitecore.Globalization;
+using Synthesis.FieldTypes.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace Synthesis.Solr.ContentSearch
+{
+	class SynthesisSolrFieldNameTranslator : FieldNameTranslator
+	{
+		private SolrSearchContext _context;
+
+		public SynthesisSolrFieldNameTranslator(SolrSearchContext context)
```

Comment:
Conversation on this just now:

(8:15:52 AM) Jeff Darchuk: i considered having the code generation generate differently for a solr instance of synthesis
(8:16:03 AM) Jeff Darchuk: with the _t on the end of the search index field name attribute
(8:16:41 AM) kfigy@chat/desktop: hmm yeah.
(8:17:19 AM) Jeff Darchuk: but i would have to defer to you to see if that would involve a solr dependency on the main synthesis project
(8:18:12 AM) kfigy@chat/desktop: if it was tweaked right, we could slot in an inherited metadata generator and change how it does indexfield, which wouldnt require a dep. lemme investigate that possibility
(8:26:22 AM) kfigy@chat/desktop: https://github.com/kamsar/Synthesis/blob/master/Source/Synthesis/Generation/MetadataGenerator.cs#L62 thats where it sets the indexfield
(8:26:48 AM) kfigy@chat/desktop: which comes from a passed in FieldNameTranslator instance
(8:27:00 AM) kfigy@chat/desktop: hypothetically, SOLR could get injected there?
(8:27:31 AM) Jeff Darchuk: yeah, i dont see why not
(8:28:02 AM) Jeff Darchuk: that would be the correct way to implement this, IMO
(8:28:25 AM) kfigy@chat/desktop: the translator is injected from the /sitecore/synthesis/providers/indexConfiguration xpath setting in config
(8:28:51 AM) Jeff Darchuk: alright then, sounds easy enough
(8:28:52 AM) kfigy@chat/desktop: which defaults to the master index
(8:29:08 AM) kfigy@chat/desktop: so hypothetically, if the master index is SOLR, it should ALREADY do it
(8:29:21 AM) kfigy@chat/desktop: did you regenerate the model after converting to solr?
(8:29:39 AM) Jeff Darchuk: yeah, i did
(8:29:59 AM) kfigy@chat/desktop: hmm. maybe some debugging is in order